dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/EventDataQueryRequest.java

```diff
@@ -256,14 +258,11 @@ public EventDataQueryRequestBuilder fromCriteria( EventsAnalyticsQueryCriteria c
                  *
                  * returns: TODAY:LAST_UPDATED;LAST_WEEK:INCIDENT_DATE
                  */
-                String customDateFilters = Arrays.stream( AnalyticsDateFilter.values() )
-                    .filter( AnalyticsDateFilter::appliesToEvents )
-                    .filter( analyticsDateFilter -> analyticsDateFilter.getEventExtractor().apply( criteria ) != null )
-                    .map( analyticsDateFilter -> String.join( DIMENSION_NAME_SEP,
-                        withOptionSeparator( analyticsDateFilter.getEventExtractor().apply( criteria ) ),
-                        analyticsDateFilter.getTimeField().name() ) )
-                    .collect( Collectors.joining( OPTION_SEP ) );
-
+                String customDateFilters = getCustomDateFilters(
+                    AnalyticsDateFilter::appliesToEvents,
+                    analyticsDateFilter -> o -> analyticsDateFilter.getEventExtractor()
+                        .apply( (EventsAnalyticsQueryCriteria) o ),
+                    criteria );
```

Comment:
I extracted a private method here which was used in both events and enrollments scenarios